packages/payment/src/StripeService.ts

```diff
@@ -0,0 +1,105 @@
+import { injectable } from 'inversify';
+import { WishCardRepository } from '@donategifts/wishcard';
+import { UserRepository } from '@donategifts/user-data';
+import { logger } from '@donategifts/helper';
+import * as mongoSanitize from 'express-mongo-sanitize';
+import Stripe from 'stripe';
+import { IDonationHook, IStripeIntent, IUser, IWishCard, TypeObjectId } from '@donategifts/common';
+import { calculateWishItemTotalPrice, handleDonation } from './helper/paymentHelper';
+
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const stripe = new Stripe(process.env.STRIPE_KEY!, {
+	apiVersion: '2020-08-27',
+});
+
+@injectable()
+export class StripeService {
+	constructor(
+		private wishCardRepository: typeof WishCardRepository = WishCardRepository,
+		private lastWishcardDonation: string = '',
+		private userRepository: typeof UserRepository = UserRepository,
+	) {}
+
+	public async createIntent(
+		stripeData: IStripeIntent,
+	): Promise<{ success: boolean; error?: string; clientSecret?: string }> {
+		const { wishCardId, email, agencyName, userDonation, userId } = stripeData;
+		// Create a PaymentIntent with the order amount and currency
+		const wishCard = await this.wishCardRepository.getWishCardByObjectId(
+			mongoSanitize.sanitize(wishCardId) as TypeObjectId<IWishCard>,
+		);
+
+		const user = await this.userRepository.getUserById(
+			mongoSanitize.sanitize(userId) as TypeObjectId<IUser>,
+		);
+
+		// By default stripe accepts "pennies" and we are storing in a full "dollars". 1$ == 100
+		// so we need to multiple our price by 100. Genious explanation
+		const PENNY = 100;
+		let totalItemPrice = await calculateWishItemTotalPrice(wishCard.wishItemPrice);
+		if (wishCard) {
+			const metadata: {
+				userDonation: number | null;
+				wishCardId: string | null;
+				userId: string | null;
+				agencyName: string | null;
+			} = {
+				wishCardId: wishCard._id.toString(),
+				userId: user._id.toString(),
+				agencyName,
+				userDonation: null,

```

Comment:
a bit pedantic but can you move userdonation to the beginning so that the order is the same in both objects